mapper/cumulocity/c8y_translator_lib/src/serializer.rs

```diff
@@ -165,13 +151,20 @@ impl GroupedMeasurementVisitor for C8yJsonSerializer {
             return Err(MeasurementStreamError::UnexpectedEndOfGroup.into());
         }
 
-        self.buffer.push('}');
+        self.json.write_close_obj();
         self.needs_separator = true;
         self.is_within_group = false;
         Ok(())
     }
 }
 
+impl C8yJsonSerializer {
+    pub fn into_string(&mut self) -> Result<String, C8yJsonSerializationError> {
+        self.end()?;
+        Ok(self.json.clone().into_string())
+    }
+}
```

Comment:
It would be better to define this method in the same code block as `C8yJsonSerializer.bytes()`.